app/timeline-menu/timeline-menu.ts

```diff
@@ -0,0 +1,63 @@
+let _ = require('lodash');
+
+export enum TimelineMenuRange { LastWeek, LastMonth, LastYear, LastDecade, PreDecade, CustomRange }
+
+export class TimelineMenu {
+  LIMIT_NUM = 8640000000000000;
+  MIN_DATE = new Date(-this.LIMIT_NUM);
+  MAX_DATE = new Date(this.LIMIT_NUM);
+  periods = [
+    { 'title': '7 ימים אחרונים', 'value': 'LastWeek', 'start': '', 'end': '' },
+    { 'title': '30 ימים אחרונים', 'value': 'LastMonth', 'start': '', 'end': '' },
+    { 'title': 'השנה', 'value': 'LastYear', 'start': '', 'end': '' },
+    { 'title': 'בעשור האחרון', 'value': 'LastDecade', 'start': '', 'end': '' },
+    { 'title': 'מלפני עשור ויותר', 'value': 'PreDecade', 'start': '', 'end': '' },
+    { 'title': 'טווח תאריכים לבחירתך', 'value': 'CustomRange', 'start': '', 'end': '' }
+  ];
+
+  constructor() {
+    this.initPeriods();
+  }
+
+  getPeriod(range: TimelineMenuRange) {
+    return _.find(this.periods, ['value', TimelineMenuRange[range]]);
+  }
+
+  private initPeriods(): any {
+    let that = this;
```

Comment:
this is TypeScript - you don't need this here.

Reply:
this here is current instance of the object. it is not exists in the async forEach function